### PR TITLE
[storage] add error counters and log for every LibraDB API

### PR DESCRIPTION
**Pull request by msmouse**
Also distinguish latency gauges for successes and failures.

## Motivation

To support error based dashboards and alerts.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

existing coverage

### CODE REVIEW

File: storage/libradb/src/lib.rs [hunk 1]
```diff
@@ -483,37 +483,31 @@ impl DbReader for LibraDB {
...
         end_epoch: u64,
     ) -> Result<EpochChangeProof> {
-        let _timer = LIBRA_STORAGE_API_LATENCY_SECONDS
-            .with_label_values(&["get_epoch_ending_ledger_infos"])
-            .start_timer();
-
-        let (ledger_info_with_sigs, more) =
-            Self::get_epoch_ending_ledger_infos(&self, start_epoch, end_epoch)?;
-        Ok(EpochChangeProof::new(ledger_info_with_sigs, more))
+        gauged_api("get_epoch_ending_ledger_infos", || {
```
Comment (davidiw) [inline storage/libradb/src/lib.rs:486]: waaah, what's the benefit of this approach versus say...

```
let _timer = start_storage_timer(name);

run_op
```

maybe one additional lines of code but the following is a little interesting... though I think safety rules also looks a little interesting now: https://github.com/libra/libra/blob/master/consensus/safety-rules/src/safety_rules.rs#L442 <3

code is fine, but figure let's chat

Comment (msmouse) [reply storage/libradb/src/lib.rs:486]: It times "Ok"s and "Err"s seperately.

Comment (davidiw) [reply storage/libradb/src/lib.rs:486]: ahhh, fancy

**Review by davidiw — APPROVED** [review]

**Pull request closed without merging**